components/custom/SearchableSelect/SearchableSelect.svelte

```diff
@@ -32,6 +33,18 @@ const onChange = (e) => {
   choice = internalChoice
   element.blur()
 }
+
+const clearChoice = () => {
+  savedChoice = choice
+  choice = ''
+}
+
+const onBlur = () => {
+  if (savedChoice && !choice) {
+    choice = savedChoice
+  }
+  dispatch('check', choice)
```

Comment:
Does `onChange` trigger before `onBlur`? I'm just wondering if `choice` will have been updated yet. Previously, on-blur we were dispatching the event's target.value, but now we're dispatching whatever value is in `choice`. It might work correctly, but I know that kind of edge-case timing can potentially cause problems.

Reply:
Good question. onChange is triggered first when a choice is selected, and we are triggering onBlur in the onChange handler. Clicking the clear button doesn't trigger onChange, just like starting to delete your query doesn't so I think we are ok here, but I will keep an eye out for any problems.